### ISSUE TITLE
Add forEachView function to WidgetModel 

### ISSUE THREAD

**Issue by ianhi** [jupyter-widgets/ipywidgets#2920]
I think that the `forEachView` function as implemented in ipycanvas by @martinRenou is super useful. Would it be possible to add this to [WidgetModel](https://github.com/jupyter-widgets/ipywidgets/blob/6be18d9b75353f7b4a1c328c6ea06d8959f978f6/packages/base/src/widget.ts#L69) class?

Relevant lines from ipycanvas
https://github.com/martinRenou/ipycanvas/blob/09ca4e04c3c43bca5df46975cad28ada3d50750e/src/widget.ts#L293-L299
https://github.com/martinRenou/ipycanvas/blob/09ca4e04c3c43bca5df46975cad28ada3d50750e/src/widget.ts#L327

I don't know exactly how the typing of the return types and the Dict of views would work but I think this would be possible?

**Comment by vidartf** [comment]
The typical pattern I use is to `this.trigger('myevent')` on the model, and have the views listen to those events. That helps me separate the model and view logic, as intended :)

**Comment by martinRenou** [comment]
I totally agree with your point @vidartf, it's a better separation of concerns.

Although the `forEachView` in ipycanvas was (if I remember correctly) there for performances purpose. I did not want to use events in this particular case, to not slow down the code.

**Comment by ianhi** [comment]
Looking back on this I agree. But this should be documented somewhere - in particular the "correct" or intended way to use the framework should be stated explicitly somewhere. Otherwise more people like me will find existing examples and think that it's the best way!
